integration-tests/common/test_common.go

```diff
@@ -176,75 +104,91 @@ type ContractsState struct {
 	OCRVault      string `json:"ocr_vault"`
 }
 
-func (m *OCRv2TestState) LabelChaosGroups() {
-	m.LabelChaosGroup(1, 5, ChaosGroupFaulty)
-	m.LabelChaosGroup(6, 19, ChaosGroupOnline)
-	m.LabelChaosGroup(0, 8, ChaosGroupYellow)
-	m.LabelChaosGroup(0, 9, ChaosGroupLeftHalf)
-	m.LabelChaosGroup(10, 19, ChaosGroupRightHalf)
-}
-
 func (m *OCRv2TestState) DeployCluster(contractsDir string) {
-	if m.Common.IsK8s {
+	if *m.Config.TestConfig.Common.InsideK8s {
 		m.DeployEnv(contractsDir)
+
+		// Setting up the URLs
+		m.Common.ChainDetails.RPCURLExternal = m.Common.Env.URLs["sol"][0]
+		m.Common.ChainDetails.WSURLExternal = m.Common.Env.URLs["sol"][1]
+
+		if *m.Config.TestConfig.Common.Network == "devnet" {
+			m.Common.ChainDetails.RPCUrl = *m.Config.TestConfig.Common.RPC_URL
+			m.Common.ChainDetails.RPCURLExternal = *m.Config.TestConfig.Common.RPC_URL
+			m.Common.ChainDetails.WSURLExternal = *m.Config.TestConfig.Common.WS_URL
+		}
+
+		m.Common.ChainDetails.MockserverURLInternal = m.Common.Env.URLs["qa_mock_adapter_internal"][0]
+		m.Common.ChainDetails.MockServerEndpoint = "five"
 	} else {
 		env, err := test_env.NewTestEnv()
-		require.NoError(m.T, err)
-		sol := test_env_sol.NewSolana([]string{env.DockerNetwork.Name})
+		require.NoError(m.Config.T, err)
+		sol := test_env_sol.NewSolana([]string{env.DockerNetwork.Name}, *m.Config.TestConfig.Common.DevnetImage, m.Common.AccountDetails.PublicKey)
 		err = sol.StartContainer()
-		require.NoError(m.T, err)
-		m.Common.SolanaUrl = sol.InternalHttpUrl
+		require.NoError(m.Config.T, err)
+
+		// Setting the External RPC url for Gauntlet
+		m.Common.ChainDetails.RPCUrl = sol.InternalHttpUrl
+		m.Common.ChainDetails.RPCURLExternal = sol.ExternalHttpUrl
+		m.Common.ChainDetails.WSURLExternal = sol.ExternalWsUrl
+
+		if *m.Config.TestConfig.Common.Network == "devnet" {
+			m.Common.ChainDetails.RPCUrl = *m.Config.TestConfig.Common.RPC_URL
+			m.Common.ChainDetails.RPCURLExternal = *m.Config.TestConfig.Common.RPC_URL
+			m.Common.ChainDetails.WSURLExternal = *m.Config.TestConfig.Common.WS_URL
+		}
+
 		b, err := test_env.NewCLTestEnvBuilder().
 			WithNonEVM().
-			WithTestInstance(m.T).
-			WithTestConfig(m.TestConfig).
+			WithTestInstance(m.Config.T).
+			WithTestConfig(m.Config.TestConfig).
 			WithMockAdapter().
 			WithCLNodeConfig(m.Common.DefaultNodeConfig()).
-			WithCLNodes(m.Common.NodeCount).
-			WithCLNodeOptions(m.Common.NodeOpts...).
+			WithCLNodes(*m.Config.TestConfig.OCR2.NodeCount).
+			WithCLNodeOptions(m.Common.TestEnvDetails.NodeOpts...).
```

Comment:
FYI we will be removing that soon in core and that node config will be part of TOML config